### ISSUE TITLE
n.extra from class Notify is empty.

### ISSUE THREAD

**Issue by randomw1zard** [denizzzka/dpq2#60]
I'm trying to use the LISTEN functionality. Notifications are working. 
I'm able to use the .name and .pid from the notification but for some odd reason i'm unable to use the .extra property. When i try to print it using writeln i get a line of emptiness.

`class Notify
{
    private immutable PGnotify* n;

    this(immutable PGnotify* pgn )
    {
        n = pgn;
        cast(void) enforceEx!OutOfMemoryError(n, "Can't write notify");
    }

    ~this()
    {
        PQfreemem( cast(void*) n );
    }

    /// Returns notification condition name
   string name() { return to!string( n.relname ); }

    /// Returns notification parameter
   string extra() { return to!string( n.extra ); } THIS DOESN"T WORK !!!!

    /// Returns process ID of notifying server process
    size_t pid() { return n.be_pid; }
`

**Comment by denizzzka** [comment]
(Please check format of code sample markdown)

> but for some odd reason i'm unable to use the .extra property. When i try to print it using writeln i get a line of emptiness.

How do you generate a notify?

**Comment by randomw1zard** [comment]
```
connect();  
con.exec( "listen project_inserted_event; notify project_inserted_event" );

sleep(2.seconds);

con.sendQuery("INSERT INTO projects(title,description,pipelines) values('test','testss','{0}')");

auto notification= con.getNextNotify();
if(notification.name == "project_inserted_event" ){	
        writeln(notification.pid);	
	writeln(notification.name);
        writeln(notification.extra);	
}
```

**Comment by denizzzka** [comment]
Hmm.. I am improved test for extra payloads and it works ok for me and Travis-CI:
https://github.com/denizzzka/dpq2/commit/368a3c54164f45ca64e82e9fad1b32e9b42d8ebb

**Comment by denizzzka** [comment]
This:

`con.exec( "listen project_inserted_event; notify project_inserted_event" );`

Here is forgotten extra payload string

**Comment by randomw1zard** [comment]
Thanks ! 2 things were stopping me from using this properly. 
first i'm an idiot i forgot to delete the notify project_inserted_event string in con.exec
it triggered the notification without a payload the insert query triggers the notification again but i already called getNextNotify so i got the old one which didn't have a payload... I removed the first trigger and the program started exiting with code 11. I added another sleep before invoking getNextNotify in order to give the program some time to process things and everything is working fine which left me with the following snippet. Works like a charm thanks for the fast answers : ))
 ```

con.exec( "listen project_inserted_event; );
sleep(2.seconds);
con.sendQuery("INSERT INTO projects(title,description,pipelines) values('test','testss','{0}')");
sleep(2.seconds);
auto notification= con.getNextNotify();
if(notification.name == "project_inserted_event" ){	
        writeln(notification.pid);	
        writeln(notification.name);
        writeln(notification.extra);	
}
```
